src/python/pants/core/util_rules/partitions.py

```diff
@@ -38,6 +38,26 @@ class PartitionerType(Enum):
     DEFAULT_SINGLE_PARTITION = "default_single_partition"
     """Registers a partitioner which returns the inputs as a single partition."""
 
+    DEFAULT_ONE_PARTITION_PER_INPUT = "default_one_partition_per_input"
+    """Registers a partitioner which returns a single-element partition per input."""
```

Comment:
You need to also state they partition key will be the address (although it might be something else, see below).